### DISCUSSION TITLE
Is there a way to reuse reducers/effects?

### DISCUSSION THREAD

**Discussion by SirPinkBeardSK** [mrpmorris/Fluxor#502] [Q&A]
If I have a reducer or effect that uses the typeOf in the annotation, is there a way to stack annotations so that the single effect or reducer would be triggered?

``` C#
[ReducerMethod(typeof(ResetStateAction))]
[ReducerMethod(typeof(ClearStateAction))]
public static TestState Reduce_ResetStateActions(TestState state) => new()
 ``` 

This is a useful feature in Ngrx with Typescript, that I'm not sure would be possible in C#.

Ideally, there might be a way to map a set of actions that both implement an interface so that the data inside the actions could be piped through the effect or reducer the same way like below.

``` C#
interface ITestDataAction
{
  string Name {get;}
 List<int> Numbers {get;}
}

public record LoadPersonalDataAction(string Name, List<int> Numbers) : ITestDataAction;
public record ChangeNameAndNumbers(string Name, List<int> Numbers): ITestDataAction;

[ReducerMethod(typeof(LoadPersonalDataAction))]
[ReducerMethod(typeof(ChangeNameAndNumbers))]
public static TestState Reduce_ResetStateActions(TestState state, ITestDataAction action) 
=> state with {Name = action.Name, Numbers = action.Numbers};
```

I've found this incredibly helpful to prevent duplicate logic and code in my reducers and effects in Ngrx, but I get that this might not be possible for c#, thought I'd ask anyway

**Comment by SirPinkBeardSK** [comment]
While not quite reusable, you can get this functionality if you use an interface on the Action for Effects. Multiple actions that use the interfaces will be processed by the same effect that is listening to the Interface

**Comment by mrpmorris** [comment]
You can only have one `ReducerMethod` per method. If you want different actions then they need to descend from a common ancestor or you need to have the `action` be a type of interface and have the action implement that.
